service/history/stateBuilder.go

```diff
@@ -94,8 +99,10 @@ func (b *stateBuilderImpl) applyEvents(domainID, requestID string, execution sha
 	var newRunStateBuilder mutableState
 	for _, event := range history {
 		lastEvent = event
-		// must set the current version, since this is standby here, not active
-		b.msBuilder.UpdateReplicationStateVersion(event.GetVersion(), true)
+		if b.msBuilder.GetReplicationState() != nil {
```

Comment:
for cross dc workflow, this check will always be true

Reply:
I will fix the comment, it is also being used in the active side.